sylvia-derive/src/message_type.rs

```diff
@@ -161,4 +162,20 @@ impl MsgType {
             },
         }
     }
+
+    #[cfg(feature = "orch")]
+    pub fn emit_derive_call(&self) -> TokenStream {
+        let sylvia = crate_module();
+        match self {
+            MsgType::Query => quote! {
+                #[derive(#sylvia ::serde::Serialize, #sylvia ::serde::Deserialize, Clone, Debug, PartialEq, #sylvia ::schemars::JsonSchema, #sylvia:: cw_schema::QueryResponses, #sylvia:: cw_orch::QueryFns)]
+            },
+            MsgType::Exec => quote! {
+                #[derive(#sylvia ::serde::Serialize, #sylvia ::serde::Deserialize, Clone, Debug, PartialEq, #sylvia ::schemars::JsonSchema, #sylvia:: cw_orch::ExecuteFns)]
+            },
+            _ => quote! {
+                #[derive(#sylvia ::serde::Serialize, #sylvia ::serde::Deserialize, Clone, Debug, PartialEq, #sylvia ::schemars::JsonSchema)]
+            },
+        }
+    }
```

Comment:
No need for such split.

Add method in `MsgType` to return `#sylvia:: cw_orch::QueryFns` or `#sylvia:: cw_orch::ExecuteFns` for `MsgType::Query/Exec` respectively if flag is enabled.